code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -23,6 +23,7 @@
 	if(changed)
 		SEND_SIGNAL(src, COMSIG_PAUSE_FLOATING_ANIM, 0.3 SECONDS)
 		animate(src, transform = ntransform, time = (lying_prev == 0 || lying_angle == 0) ? 2 : 0, pixel_y = final_pixel_y, dir = final_dir, easing = (EASE_IN|EASE_OUT))
+	return ..()
```

Comment:
this is inconsistent. In /mob/living/carbon/alien/larva/update_transform(), you call the parent before executing the child's code and in this proc, you are calling the parent after the child code has executed. If you are looking to get the results of a transformation from update_transform, then you may need to wrap it in another proc where a new signal id can be sent after transformation is complete.